src/platforms/hubspot/schemas.ts

```diff
@@ -0,0 +1,655 @@
+  groupName: z.string(),
+  options: z.array(hubspotPropertyOptionSchema).optional(),
+});
+export const hubspotCustomPropertyCreateSchema = z.object({
+  objectType: hubspotModuleSchema,
+  property: hubspotCustomPropertySchema,
+});
+export type HubspotCustomPropertyCreate = z.infer<
+  typeof hubspotCustomPropertyCreateSchema
+>;
+
+export const hubspotPropertySchema = z
+  .object({
+    name: z.string(),
+    label: z.string(),
+    type: hubspotPropertyTypeSchema,
+    fieldType: hubspotPropertyFieldTypeSchema,
+    hubspotDefined: z.boolean(),
+    options: z.array(hubspotPropertyOptionSchema).optional(),
+    modificationMetadata: z.object({
+      readOnlyValue: z.boolean(),
+    }),
+  })
+  .passthrough();
+export type HubspotProperty = z.infer<typeof hubspotPropertySchema>;
+
+// -
+// Associations
+// -
+export const HUBSPOT_DEFINED_CATEGORY = 'HUBSPOT_DEFINED';
+export const hubspotAssociationCreateSchema = z.object({
+  fromId: hubspotIdSchema,
+  fromType: hubspotModuleSchema,
+  toId: hubspotIdSchema,
+  toType: hubspotModuleSchema,
+  category: z.string().optional(),
+  typeId: hubspotIdSchema.optional(),
+});
+export type HubspotAssociationCreate = z.infer<
+  typeof hubspotAssociationCreateSchema
+>;
+export const hubspotAssociationResponseSchema = z
+  .object({
+    from: z.object({
+      id: hubspotIdSchema,
+    }),
+    to: z.object({
+      id: hubspotIdSchema,
+    }),
+    associationSpec: z
+      .object({
+        associationCategory: z.string(),
+        associationTypeId: hubspotIdSchema,
+      })
+      .passthrough(),
+  })
+  .passthrough();
+export type HubspotAssociationResponse = z.infer<
+  typeof hubspotAssociationResponseSchema
+>;
+export const hubspotAssociationLabelInputSchema = z.object({
+  fromType: hubspotModuleSchema,
+  toType: hubspotModuleSchema,
+});
+export type HubspotAssociationLabelInput = z.infer<
+  typeof hubspotAssociationLabelInputSchema
+>;
+export const hubspotAssociationLabelOutputSchema = z
+  .object({
+    category: z.string(),
+    typeId: hubspotIdSchema,
+  })
+  .passthrough();
+export type HubspotAssociationLabelOutput = z.infer<
+  typeof hubspotAssociationLabelOutputSchema
+>;
+
+// -
+// Client Definitions
+// -
+export type FindObjectInput = {
+  id: string;
+  associations?: HubspotCommonAssociations[];
+};
+export type ListObjectInput = {
+  after?: string;
+  pageSize?: number;
+  associations?: HubspotCommonAssociations[];
+};
+export type ListOutput<T> = {
+  paging?: {
+    next?: {
+      after?: string;
+    };
+  };
+  results?: T[];
+};
+type ClientAction<InputType, OutputType> = (
+  auth: Auth,
+  args: InputType,
+) => Promise<ClientResult<OutputType>>;
+
+type CrudOperations<T, TCreate, TUpdate> = {
+  find: ClientAction<FindObjectInput, T>;
+  list: ClientAction<ListObjectInput, ListOutput<T>>;
+  create: ClientAction<TCreate, T>;
+  update: ClientAction<TUpdate, T>;
+};
+
+export interface HubspotClient extends PlatformClient {
```

Comment:
I had this here since it was helpful while developing, but I think it's better to not use this and just use `typeof` for the client. Will remove

Reply:
Are you planning to do this before merging?